src/mlang/m_ir/mir_typechecker.ml

```diff
@@ -75,58 +65,46 @@ let rec typecheck_top_down (ctx : ctx) (e : expression Pos.marked) : ctx =
           ]
 
 and typecheck_func_args (f : func) (pos : Pos.t) :
-    ctx -> Mir.expression Pos.marked list -> ctx =
+    bool -> Mir.expression Pos.marked list -> unit =
   match f with
   | SumFunc | MinFunc | MaxFunc ->
-      fun ctx args ->
+      fun in_generic_table args ->
```

Comment:
Is there a reason why we keep different `fun in_generic_table args` in the different cases of the `match`?

Reply:
I think not, I factorized all this (among other things).